File: stack.py
'''
FILO | LIFO
'''

class Stack:
	def __init__(self):
		self.stack = []
		self.count = 0
	def push(self, value):
		self.stack.append(value)
		self.count += 1
	def pop(self):
		if self.count == 0 or len(self.stack) == 0:
			return None
		else:
			self.count -= 1
			return self.stack.pop(-1)
	def peel(self):
		if self.count == 0 or len(self.stack) == 0:
			return None
		else:
			return self.stack[-1]
	def size(self):
		return self.count
	def isEmpty(self):
		return self.count == 0

'''
#Test cases
stack = Stack()
stack.push(1)
stack.push(2)
stack.push(4)
stack.push(6)
stack.push(8)
stack.push(10)
stack.push(12)
stack.push(14)
print stack.pop(), 14
print stack.size(), 6
print stack.pop(), 12
print stack.pop(), 10
print stack.pop(), 8
print stack.pop(), 6
print stack.pop(), 4
print stack.pop(), 2
print stack.peel(), 1
print stack.size(), 1
print stack.isEmpty(), True
'''

File: java_version/Stack.java
import java.io.*;
import java.util.*;

class Stack{

    private List<String> stack;
    private int count;

    public Stack(){
        this.stack = new ArrayList<String>();
        this.count = 0;
    }

    public void push(String value){
        this.stack.add(value);
        this.count++;
    }

    public String pop(){
        if(this.count > 0){
            this.count--;
            return this.stack.remove(0);
        }else{
            return null;
        }
    }

    public String peel(){
        if(this.count == 0){
            return null;
        }else{
            return this.stack.get(0);
        }
    }

    public int getSize(){
        return this.count;
    }

    public static void main(String[] args){
        List<String> strList = new ArrayList<String>(Arrays.asList(args[0].split(",")));
        Stack stack = new Stack();
        for(String str : strList){
            stack.push(str);
        }
        while(stack.getSize() != 0){
            System.out.println(stack.peel());
            System.out.println(stack.pop());
        }
    }
}
